apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -187,6 +192,46 @@ function pollPaperStatus(): InvokeConfig<Context, PaperHandlerStatusEvent> {
   };
 }
 
+function buildAuthStatusObservable() {
+  return ({ auth, authPollingIntervalMs, workspace }: Context) => {
+    return timer(0, authPollingIntervalMs).pipe(
+      switchMap(async () => {
+        try {
+          const authStatus = await auth.getAuthStatus(
+            constructAuthMachineState(workspace)
+          );
+          debug('auth status polled');
+          if (isCardlessVoterAuth(authStatus)) {
+            debug('emit cardless voter auth event');
+            return { type: 'AUTH_STATUS_CARDLESS_VOTER' };
+          }
+          if (isPollWorkerAuth(authStatus)) {
+            debug('emit poll worker auth event');
+            return { type: 'AUTH_STATUS_POLL_WORKER' };
+          }
+
+          debug('Unhandled auth status in observable: %O', authStatus);
+          return { type: 'AUTH_STATUS_UNHANDLED' };
+        } catch (err) {
+          debug('Error in auth observable: %O', err);
+          return { type: 'AUTH_STATUS_UNHANDLED' };
+        }
+      }),
+      timeout({
+        each: PAPER_HANDLER_STATUS_POLLING_TIMEOUT_MS,
+        with: () => throwError(() => new Error('paper_status_timed_out')),
+      })
+    );
+  };
+}
+
+function pollAuthStatus(): InvokeConfig<Context, PaperHandlerStatusEvent> {
```

Comment:
I don't love adding more polling, but this seemed cleaner than expecting the frontend to "watch" auth changes and change the backend state machine state via `useEffect`

Reply:
Agree that polling on the backend is cleaner